vault/external_tests/identity/login_mfa_okta_test.go

```diff
@@ -6,20 +6,56 @@ package identity
 import (
 	"context"
 	"fmt"
+	"os"
 	"reflect"
+	"strings"
 	"testing"
+	"time"
 
 	"github.com/hashicorp/vault/api"
+	"github.com/hashicorp/vault/helper/constants"
+	"github.com/hashicorp/vault/helper/testhelpers"
+	logicaltest "github.com/hashicorp/vault/helper/testhelpers/logical"
 	"github.com/hashicorp/vault/helper/testhelpers/minimal"
 )
 
-var (
-	org_name  = "<okta org name>"
-	api_token = "<okta api token>"
-)
+// To run these tests, set the following env variables:
+// VAULT_ACC=1
+// OKTA_ORG=dev-219337
+// OKTA_API_TOKEN=<generate via web UI, see Confluence for login details>
+// OKTA_USERNAME=test3@example.com
+// OKTA_PASSWORD=<find in 1password>
+//
+// You will need to install the Okta client app on your mobile device and
+// setup MFA in order to use the Okta web UI. This test does not exercise
+// MFA however (which is an enterprise feature), and therefore the test
+// user in OKTA_USERNAME should not be configured with it. Currently,
+// test3@example.com is not a member of testgroup, which is the group with
+// the profile that requires MFA. If you need to use a different group name
+// for the test group, you can set:
+// OKTA_TEST_GROUP=alttestgroup
+//
+// To test with Okta TOTP (instead of Okta push verify), set:
+// OKTA_USE_TOTP=1
 
 func TestOktaEngineMFA(t *testing.T) {
-	t.Skip("This test requires manual intervention and OKTA verify on cellphone is needed")
+	if os.Getenv(logicaltest.TestEnvVar) == "" {
+		t.Skip("This test requires manual intervention and OKTA verify on cellphone is needed")
+	}
+	if os.Getenv("OKTA_USE_TOTP") == "1" {
+		// NB: If the Okta Org has push verify disabled, this test would fail without this Skip.
+		t.Skip("This test uses Okta push for MFA, but OKTA_USE_TOTP is set and this test cannot pass a TOTP.")
+	}
+
+	// Ensure each cred is populated.
+	credNames := []string{
+		"OKTA_ORG",
+		"OKTA_API_TOKEN",
+		"OKTA_USERNAME",
+		"OKTA_PASSWORD",
+	}
+	testhelpers.SkipUnlessEnvVarsSet(t, credNames)
```

Comment:
I got this pattern from `builtin/credential/okta/backend_test.go`
https://github.com/hashicorp/vault/blob/58a5b77bc6f0a5ac47e3e3d6020148b585f7213a/builtin/credential/okta/backend_test.go#L43-L49